subgraphs/cryptopunk/src/mappings/cryptoPunkMarket.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  PunkOffered,
+  PunkBidEntered,
+  PunkBought,
+} from "../../generated/cryptopunkContract/cryptopunkContract";
+import { _User } from "../../generated/schema";
+import { getSellerAddressFromPunksOfferedForSale } from "../constants/utils";
+import { createBid } from "../constants/initializers";
+import { updateTrades } from "../modules/Trades";
+import * as constants from "../constants/constants";
+import { createUserCollectionAccount } from "../constants/initializers";
+export function handlePunkOffered(event: PunkOffered): void {
+  let sellerAddress = getSellerAddressFromPunksOfferedForSale(
+    event.params.punkIndex
+  );
+  createUserCollectionAccount(constants.TradeType.SELLER, sellerAddress);
+}
+
+export function handlePunkBidEntered(event: PunkBidEntered): void {
+  let tokenId = event.params.punkIndex;
+  let bidAmount = event.params.value;
+  let bidderAddress = event.params.fromAddress;
+  createUserCollectionAccount(constants.TradeType.BUYER, bidderAddress);
```

Comment:
i don't think you should increase the `buyerCount` for a collection for an entered bid (unless the trade actually went through)
- 1000 unique bids on a cryptopunk would mean 1000 buyers with this logic